src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionFactory.cs

```diff
@@ -39,7 +39,7 @@ override protected DbConnectionInternal CreateConnection(DbConnectionOptions opt
             SqlInternalConnection result = null;
             SessionData recoverySessionData = null;
 
-            SqlConnection sqlOwningConnection = owningConnection as SqlConnection;
+            SqlConnection sqlOwningConnection = (SqlConnection)owningConnection;
```

Comment:
> Whatever, there're two casts like this in the code below this code 
I agree with the changes you have made to the code below and those changes are needed. I was addressing the change for this particular line. If you don't cast to a SqlConnection, the code will still function with the below changes within reasonable limits. 

This code could have been left as is. 
In case a `sqlOwningConnection` is `null` your changes in the following line will take care of it.
If `sqlOwningConnection` is `null` then a few connection parameters will not be honored in the connection establishment and the function can still return a connection based on the input DbConnection.